adapters/Unity/UnityAdapter/GADMUnityInitializer.m

```diff
@@ -0,0 +1,195 @@
+#import "GADMediationAdapterUnity.h"
+
+@interface GADMAdapterUnity (){
+  /// Connector from Google Mobile Ads SDK to receive ad configurations.
+  __weak id<GADMAdNetworkConnector> _networkConnector;
+  
+  /// Game ID of Unity Ads network.
+  NSString *_gameID;
+  
+  /// Placement ID of Unity Ads network.
+  NSString *_placementID;
+  
+  /// Unity Ads Banner wrapper
+  GADMAdapterUnityBannerAd *_bannerAd;
+  
+  /// Unity Ads Interstitial Ad wrapper
+  GADMUnityInterstitialAd *_interstitialAd;
+  
+}
+
+@end
+
+@implementation GADMAdapterUnity
+
++ (nonnull Class<GADMediationAdapter>)mainAdapterClass {
+  return [GADMediationAdapterUnity class];
+}
+
++ (NSString *)adapterVersion {
+  return kGADMAdapterUnityVersion;
+}
+
++ (Class<GADAdNetworkExtras>)networkExtrasClass {
+  return Nil;
+}
+
+- (void)stopBeingDelegate {
+  if (_bannerAd != nil) {
+    [_bannerAd stopBeingDelegate];
+  }
+}
+
+- (void)initializeWithGameID:(NSString *)gameID withInitDelegate:(id)initDelegate{
+  if (![UnityAds isSupported]) {
+    NSString *message =
+      [[NSString alloc] initWithFormat:@"%@ is not supported for this device.",
+       NSStringFromClass([UnityAds class])];
+    [initDelegate initializationFailed:(kUnityInitializationErrorInternalError) withMessage:message];
+    return;
+  }
+  
+  if ([UnityAds isInitialized]) {
+    NSLog(@"Unity Ads has already been initialized.");
+    [initDelegate initializationComplete];
+    return;
+  }
+  
+  // Metadata needed by Unity Ads SDK before initialization.
+  GADMUnityConfigureMediationService();
+  // Initializing Unity Ads with |gameID|.
+  [UnityAds initialize:gameID testMode:NO enablePerPlacementLoad:YES initializationDelegate:initDelegate];
+}
+
+#pragma mark Interstitial Methods
+
+- (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)connector {
+  if (!connector) {
+    return nil;
+  }
+  self = [super init];
+  if (self) {
+    _networkConnector = connector;
+  }
+  return self;
+}
+
+- (void)getInterstitial {
+  id<GADMAdNetworkConnector> strongConnector = _networkConnector;
+  if (!strongConnector) return;
+  _gameID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityGameID] copy];
+  _interstitialAd = [[GADMUnityInterstitialAd alloc] initWithGADMAdNetworkConnector:strongConnector adapter:self];
+  if (!_interstitialAd) {
+    NSString *description = [NSString
+                             stringWithFormat:@"%@ initialization failed!", NSStringFromClass([GADMUnityInterstitialAd class])];
+    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(GADMAdapterUnityErrorAdObjectNil, description);
+    [strongConnector adapter:self didFailAd:error];
+    return;
+  }
+  [_interstitialAd getInterstitial];
+}
+
+- (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewController {
+  id<GADMAdNetworkConnector> strongConnector = _networkConnector;
+  if (strongConnector) {
+    [strongConnector adapterWillPresentInterstitial:self];
+  }
```

Comment:
Remove. This is causing multiple `willPresentInterstitial` callbacks when trying to present an interstitial ad.
Additionally, `_interstitialAd` is already forwarding this callback.